compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrSourceCompilerForInline.kt

```diff
@@ -58,11 +58,11 @@ class IrSourceCompilerForInline(
 
     override val inlineCallSiteInfo: InlineCallSiteInfo
         get() {
-            //TODO: support nested inline calls
+            val root = generateSequence(codegen) { it.inlinedInto }.last()
```

Comment:
If I understand  correctly this change should affect only `smap` related tests but no one of them is unmuted(

Reply:
`InlineCallSiteInfo` is also used when adding OUTERCLASS to regenerated classes. For example, in `inlineChain.kt` we have

```
val res2 = call {
    call {
        test { "OK" }
    }
}
```

Here `test` is an inline function that contains an anonymous object that calls an inline lambda, so that object is regenerated. Since `test` is inlined into a lambda defined in `_2Kt.box`, we want it to have `OUTERCLASS _2Kt box()`. `codegen` points to `ExpressionCodegen` for the inner lambda, then its `inlinedInto` is the outer lambda, and finally that one's `inlinedInto` is `box`, which has `inlinedInto == null`.

As for SMAP tests, I think quite a lot of the muted ones depend on objects in lambdas being regenerated...

Reply:
Thank you for clarifications!